config/configmapprovider/expand.go

```diff
@@ -17,23 +17,21 @@ package configmapprovider // import "go.opentelemetry.io/collector/config/config
 import (
 	"context"
 	"os"
-
-	"go.opentelemetry.io/collector/config"
 )
 
 type expandMapProvider struct {
-	base config.MapProvider
+	base Provider
 }
 
-// NewExpandMapProvider returns a config.MapProvider, that expands all environment variables for a
-// config.Map provided by the given config.MapProvider.
-func NewExpandMapProvider(base config.MapProvider) config.MapProvider {
+// NewExpandMapProvider returns a Provider, that expands all environment variables for a
+// config.Map provided by the given Provider.
+func NewExpandMapProvider(base Provider) Provider {
```

Comment:
s/NewExpandMapProvider/NewExpand?

What do you think? Same everywhere.

Reply:
`NewFile` is the only one that may sound a bit weird, but in general it sounds better with your proposed change. Will do the change everywhere so that we see how it looks and can decide